### DISCUSSION TITLE
How to add custom attribute to vehicleAttributes in opentcs 5.7.0 model-0.0.4.xsd?

### DISCUSSION THREAD

**Discussion by BenYaho** [openTCS/opentcs#84] [Q&A]
I'm using opentcs 5.7.0 and trying to add a custom attribute "ip" to the vehicle element's vehicleAttributes. However, when I try to validate the XML, I'm getting the following error:

[org.xml.sax.SAXParseException; lineNumber: 0; columnNumber: 0; cvc-complex-type.3.2.2: Attribute 'ip' is not allowed to appear in element 'vehicleAttributes'.]

I have checked the Model-0.0.4.xsd file and confirmed that the new content is in line with the XML format

How can I update the schema in model-0.0.4.xsd to allow adding custom attributes? Are there any other constraints that need to be adjusted?
[vehicleAttributes.zip](https://github.com/openTCS/opentcs/files/10997578/vehicleAttributes.zip)

**Comment by swltr** [comment]
Changing the schema is not recommended. For adding generic/custom data to the plant model, the supported way is to add key-value properties to the respective element. (See the [user's guide](https://www.opentcs.org/docs/5.8/user/opentcs-users-guide.html#_generic_properties) on this.) Example in the demo plant model:

https://github.com/openTCS/opentcs/blob/0886b5a7708ec3051d62448a2e15b3d8b9bbf1cb/openTCS-ModelEditor/src/dist/data/Demo-01.xml#L590

**Comment by BenYaho** [comment]
Thanks for the answer.
